src/server/actions/Volunteers.ts

```diff
@@ -0,0 +1,22 @@
+import dbConnect from '@/utils/db-connect';
+import VolunteerSchema from '@/server/models/Volunteer';
+import { VolunteerEntity } from '@/types/dataModel/volunteer';
+
+/**
+ * Soft delete a volunteer
+ * @param volunteerId The Id of the volunteer to be deleted
+ * @returns The volunteer in the database before soft-deletion, or null
+ */
+export async function softDeleteVolunteer(
+  volunteerId: string
+): Promise<VolunteerEntity | null> {
+  await dbConnect();
+
+  const res: VolunteerEntity | null = await VolunteerSchema.findByIdAndUpdate(
+    volunteerId,
+    { softDelete: true },
+    { strict: false } //Makes sure to add a softDelete field to the volunteer if it does not already have one
```

Comment:
Looks like we had a miss when setting up the model for this. Can you add it to the `VolunteerSchema` model and have it be required and default to false? That should remove the need for this line